### PR TITLE
Update NuGet packages to latest versions

### PR DESCRIPTION
**Pull request by olegsych**
This repository was flagged for using unsecure Newtonsoft.Json package version through the xunit package dependency. In addition to the xunit packages, upgrade all packages to their latest stable versions.

Dependencies of the shipping product binaries:
- Microsoft.ServiceFabric.Services.Remoting
  - System.Reflection.Emit and System.Reflection.Emit.ILGeneration from 4.3.0 to 4.7.0
  - System.Diagnostics.DiagnosticSource from 4.3.0 to 8.0.1

Dependencies of the test projects:
- FluentAssertions from 5.9.0 to 6.12.0
- Moq from 4.13.0 to 4.20.70
- Microsoft.NET.Test.Sdk from 15.9.0 to 17.10.0
- xunit from 2.4.1 to 2.8.1
- xunit.runner.visualstudio from 2.4.1 to 2.8.1
- Microsoft.Diagnostics.Tracing.TraceEvent from 2.0.45 to 3.1.12
- System.Net.Http.WinHttpHandler from 6.0.0 to 8.0.0

### CODE REVIEW

**Comment by olegsych** [comment]
@pakunapamsft, @gobradovic-MSFT could you review this?

File: test/unittests/Microsoft.ServiceFabric.Services.Remoting.Tests/Microsoft.ServiceFabric.Services.Remoting.Tests.csproj [hunk 1]
```diff
@@ -6,14 +6,13 @@
     <AssemblyName>Microsoft.ServiceFabric.Services.Remoting.Tests</AssemblyName>
     <RootNamespace>$(AssemblyName)</RootNamespace>
    <TargetFramework>net6.0</TargetFramework>
-   <DocumentationFile>$(OutputPath)\$(AssemblyName).xml</DocumentationFile>
```
Comment (gobradovic-MSFT) [inline test/unittests/Microsoft.ServiceFabric.Services.Remoting.Tests/Microsoft.ServiceFabric.Services.Remoting.Tests.csproj:9]: I'm curious, why is this removed?

**Review by gobradovic-MSFT — APPROVED** [review]

File: test/unittests/Microsoft.ServiceFabric.Services.Remoting.Tests/Microsoft.ServiceFabric.Services.Remoting.Tests.csproj (hunk 1 above)
Comment (olegsych) [reply test/unittests/Microsoft.ServiceFabric.Services.Remoting.Tests/Microsoft.ServiceFabric.Services.Remoting.Tests.csproj:9]: Good question. I should have mentioned it. Presence of the DocumentationFile property triggers additional compiler warnings about doc comments missing from the public types or methods. I really like these warnings in prod code because it reminds you to document public APIs, but in the test code it's an overkill, and I think it actually encourages bad naming habits by forcing you to supply a comment that duplicates the type or member name. Converting tests from `void` to `async Task` started generating a lot of warnings, which seems completely useless in test code.

File: src/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.csproj [hunk 2]
```diff
@@ -11,7 +11,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.ServiceFabric.FabricTransport.Internal" Version="$(NugetPkg_Version_Microsoft_ServiceFabric_FabricTransport_Internal)" />
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.3.0" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="8.0.1" />
```
Comment (pakunapamsft) [inline src/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.csproj:14]: @olegsych  - Do you think we need to release note these changes? Imagine user code as well taking dependency on these libraries and their code now throws compiler warnings, if the bindings redirects is not inclusive of the version we moved to.

**Review by pakunapamsft — APPROVED** [review]
Provided a comment, PTAL.

File: src/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.csproj (hunk 2 above)
Comment (olegsych) [reply src/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.csproj:14]: We don't have many packages and don't change them often, so it's not a bad thing to call out. I'm not sure how other package authors communicate these changes though. I imagine that teams that ship dozens of packages frequently would want to just point users to the dependencies listed on NuGet.org.

Comment (olegsych) [reply src/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.csproj:14]: I'll add release notes to the ADO work item.
